tests/tests/tx/log_syscall.rs

```diff
@@ -0,0 +1,145 @@
+use crate::helpers::TestSetupBuilder;
+use fuel_core_types::{
+    fuel_asm::*,
+    fuel_tx::*,
+};
+use rand::{
+    Rng,
+    SeedableRng,
+    rngs::StdRng,
+};
+use std::array;
+
+#[tokio::test]
+async fn syscall_logs_produced_by_both_predicates_and_script() {
```

Comment:
What is the SUT of these tests? is it `estimate_predicates` or `client.submit_and_await_commit` or both?

Reply:
Can we
- somehow ensure that the expected logs are present?
- add some failing tests here as well. Idk if this the right place for all of them, especially the `IgnoreEcal` tests, but don't think I saw those anywhere

Reply:
Makes sense. I renamed the tests in df53196c09.